AspNetCoreApplication/Repositories/BookCategoryRepository.cs

```diff
@@ -48,5 +55,17 @@ public async Task Delete(int bookId, int categoryId)
             await dataContext.SaveChangesAsync();
         }
 
+        public async Task CheckRole(int id)
+        {
+            var currentUser = authenticationService.CurrentUser;
+            var bookDetail = await base.GetByIdOrThrow(id);
+
+            var listRole = new Role[2] { Role.Admin, Role.Manager };
+            if (!listRole.Contains(currentUser.Role)
+                && bookDetail.AuthorId != currentUser.Id)
+            {
+                throw new UnauthorizedException("Không có quyền truy cập");
+            }
```

Comment:
Không nên để cái điều kiện if phức tạp

```suggestion
           if(currentUser.Role == Role.Admin) return;
           if(currentUser.Role ==  Role.Manager) return;
           
            if (bookDetail.AuthorId != currentUser.Id)
            {
                throw new UnauthorizedException("Bạn không có quyền hạn thay đổi sách này.");
            }
```

Reply:
done